README.md

```diff
@@ -15,6 +15,127 @@ with [Model Distribution](./pkg/distribution) and the
 mimics its integration with Docker Desktop and allows the package to be run in a
 standalone mode.
 
+## Contributing - Building from Source
+
+This guide is for external contributors who want to build and test the complete Docker Model Runner ecosystem from source.
+
+### Architecture Overview
+
+The Docker Model Runner ecosystem consists of three main components:
+
+- **[model-runner](https://github.com/docker/model-runner)** (this repository) - The backend daemon/server that manages and runs AI models
+- **[model-cli](https://github.com/docker/model-cli)** - The CLI client that communicates with model-runner
+- **[model-spec](https://github.com/docker/model-spec)** - The specification for packaging models as OCI artifacts
+
+### Prerequisites
+
+Before building from source, ensure you have the following installed:
+
+- **Go 1.24+** - Required for building both model-runner and model-cli
+- **Git** - For cloning repositories
+- **Make** - For using the provided Makefiles
+- **Docker** (optional) - For building and running containerized versions
+- **CGO dependencies** - Required for model-runner's GPU support:
+  - On macOS: Xcode Command Line Tools (`xcode-select --install`)
+  - On Linux: gcc/g++ and development headers
+  - On Windows: MinGW-w64 or Visual Studio Build Tools
+
+### Building the Complete Stack
+
+#### Step 1: Clone and Build model-runner (Server/Daemon)
+
+```bash
+# Clone the model-runner repository
+git clone https://github.com/docker/model-runner.git
```

Comment:
I'd leave out the clone personally, people might want to clone via ssh. And it's just generic knowledge that applies to and git based project,